mobile/src/features/Discover/common/useConfirmConnection.ts

```diff
@@ -42,17 +42,17 @@ export const useConfirmConnection = () => {
         }
 
         if (!selectedDapp) {
-          let shouldOpenMainModal = false
+          const shouldOpenMainModalRef = {current: false}
```

Comment:
I think this should be actually a ref.

Reply:
Sort of, I think the idea here is sending an object with the value (current) inside, so that the closure passed to the modal access the correct reference instead of passing it a value that doesn't update.
But not sure an actual React reference is needed. It would do the same though.

Reply:
Basically the bug right now is that `onClose` and `onConfirm` are each receiving a different copy of shouldOpenMainModal because it's just a value. By moving the value inside an object, they both can reference the inside value instead.